drivers/azure/azureutil/azureutil.go

```diff
@@ -429,9 +429,16 @@ func (a AzureClient) DeleteVirtualMachineIfExists(resourceGroup, name string) er
 	}
 
 	// Remove disk
-	if vmRef.VirtualMachineProperties != nil {
-		vhdURL := to.String(vmRef.VirtualMachineProperties.StorageProfile.OsDisk.Vhd.URI)
-		return a.removeOSDiskBlob(resourceGroup, name, vhdURL)
+	if vmProperties := vmRef.VirtualMachineProperties; vmProperties!= nil {
+		// TODO: remove unattached managed disk, requires azure sdk upgrade
+		if managedDisk := vmProperties.StorageProfile.OsDisk.ManagedDisk; managedDisk != nil {
```

Comment:
This can be updated to `if vmProperties.StorageProfile.OsDisk.ManagedDisk != nil {`